app/lib/pre_assembly/batch.rb

```diff
@@ -121,14 +123,17 @@ def process_digital_objects
         log "Completed #{dobj.druid}"
         File.open(progress_log_file, 'a') { |f| f.puts progress.to_yaml }
       end
-      log "**WARNING**: #{num_no_file_warnings} objects had no files" if num_no_file_warnings > 0
-      log "**WARNING**: #{num_failures} objects had errors during pre-assembly" if num_failures > 0
-      log "#{total_obj} objects pre-assembled"
+      errors << "#{num_no_file_warnings} objects had no files" if num_no_file_warnings > 0
+      errors << "#{num_failures} objects had errors during pre-assembly" if num_failures > 0
+      errors.each { |error| log "**WARNING**: #{error}" }
+      @had_errors = !errors.size.zero? # indicate if we had any errors
+      @error_message = errors.join(', ') # set the error message so they can be saved in the job_run
 
-      num_failures == 0 # if no failures in any objects, return true, else false
+      log "#{total_obj} objects pre-assembled"
     end
```

Comment:
we will now store the error messages and if any errors occurred in the accessor so the caller of this method can introspect and see what happened